roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3558 @@
+    // object " << this->_observables.at(0) << std::endl;
+    if (strcmp(obsname, obs->GetName()) != 0) {
+      //  coutW(ObjectHandling) << " name of existing observable " <<
+      //  this->_observables.at(0)->GetName() << " does not match expected name
+      //  " << obsname << std::endl ;
+    }
+  }
+
+  TString sbw = Form("binWidth_%s", makeValidName(obs->GetName()).Data());
+  RooRealVar *binWidth = new RooRealVar(sbw.Data(), sbw.Data(), 1.);
+  double bw = obs->numBins() / (obs->getMax() - obs->getMin());
+  binWidth->setVal(bw);
+  binWidth->setConstant(true);
+  this->_binWidths.add(*binWidth);
+
+  return obs;
+}
+
+//#ifndef USE_MULTIPRECISION_LC
+//#pragma GCC diagnostic push
+//#pragma GCC diagnostic ignored "-Wunused-parameter"
+//#endif
+
+////////////////////////////////////////////////////////////////////////////////
+/// update sample weight (-?-)
+
+inline void RooLagrangianMorphFunc::updateSampleWeights() {
+  //#ifdef USE_MULTIPRECISION_LC
+  int sampleidx = 0;
+  auto cache = this->getCache(_curNormSet);
+  const size_t n(size(cache->_inverse));
+  for (auto sampleit : this->_config.getParamCards()) {
+    const std::string sample(sampleit.first);
+    // build the formula with the correct normalization
+    LinearCombination *sampleformula = dynamic_cast<LinearCombination *>(
+        this->getSampleWeight(sample.c_str()));
+    if (!sampleformula) {
+      // coutE(ObjectHandling) << Form("unable to access formula for sample
+      // '%s'!",sample.c_str()) << std::endl;
+      return;
+    }
+    cxcoutP(ObjectHandling)
+        << "updating formula for sample '" << sample << "'" << std::endl;
+    for (size_t formulaidx = 0; formulaidx < n; ++formulaidx) {
+      const SuperFloat val(cache->_inverse(formulaidx, sampleidx));
+      if (val != val) {
+        coutE(ObjectHandling) << "refusing to propagate NaN!" << std::endl;
+      }
+      cxcoutP(ObjectHandling) << "   " << formulaidx << ":"
+                              << sampleformula->getCoefficient(formulaidx)
+                              << " -> " << val << std::endl;
+      sampleformula->setCoefficient(formulaidx, val);
+      assert(sampleformula->getCoefficient(formulaidx) == val);
+    }
+    sampleformula->setValueDirty();
+    ++sampleidx;
+  }
+  //#else
+  //  ERROR("updating sample weights currently not possible without boost!");
+  //#endif
+}
+//#ifndef USE_MULTIPRECISION_LC
+//#pragma GCC diagnostic pop
+//#endif
+
+////////////////////////////////////////////////////////////////////////////////
+/// read the parameters from the input file
+
+void RooLagrangianMorphFunc::readParameters(TDirectory *f) {
+  auto params =
+      readValues<double>(f, this->_config.getFolderNames(), "param_card", true);
+  auto flags =
+      readValues<int>(f, this->_config.getFolderNames(), "flags", false);
+  this->_config.setParamCards(params);
+  this->_config.setFlagValues(flags);
+}
```

Comment:
```suggestion
void RooLagrangianMorphFunc::Config::readParameters(TDirectory *f) {
  readValues<double>(_paramCards, f, _folderNames, "param_card", true);
  readValues<int>(_flagValues, f, _folderNames, "flags", false);
}
```
We should better fill these maps in-place to we avoid the `std::map` copy constructor which caused trouble on macOS.